moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -163,36 +173,44 @@ void planning_scene_monitor::CurrentStateMonitor::stopStateMonitor()
 {
   if (state_monitor_started_)
   {
-    joint_state_subscriber_.shutdown();
+    joint_state_subscriber_.reset();
     if (tf_buffer_ && tf_connection_)
     {
-      tf_buffer_->_removeTransformsChangedListener(*tf_connection_);
+      // TODO (anasarrak): replace this for the appropiate function, there is no similar
+      // function in ros2/geometry2.
+      // tf_buffer_->_removeTransformsChangedListener(*tf_connection_);
       tf_connection_.reset();
     }
-    ROS_DEBUG("No longer listening for joint states");
+    RCLCPP_DEBUG(node_->get_logger(), "No longer listening for joint states");
     state_monitor_started_ = false;
   }
 }
 
-std::string planning_scene_monitor::CurrentStateMonitor::getMonitoredTopic() const
+std::string planning_scene_monitor::CurrentStateMonitor::getMonitoredTopic(const std::string& joint_states_topic)
 {
+  std::string result;
+  // joint_state_subscriber_ = node_->create_subscription<sensor_msgs::msg::JointState>(joint_states_topic,
+  //                   std::bind(&planning_scene_monitor::CurrentStateMonitor::jointStateCallback, this,
+  //                   std::placeholders::_1));
   if (joint_state_subscriber_)
-    return joint_state_subscriber_.getTopic();
+    result = joint_state_subscriber_->get_topic_name();
   else
-    return "";
+    result = std::string("");
```

Comment:
This case can be removed if result is initialized empty. Also, returning result directly would be even shorter